parachain/pallets/eth-app/src/lib.rs

```diff
@@ -141,13 +142,19 @@ pub mod pallet {
 			sender: H160,
 			recipient: <T::Lookup as StaticLookup>::Source,
 			amount: U256,
+			para_id: u32,
```

Comment:
Lets rather make `para_id` an `Option<u32>`, so we explicitly know if the Solidity side passed a para_id or not. See my equivalent comments on the solidity side.